test/jdk/javax/swing/JViewport/ScrollRectToVisibleTest3.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 1999, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @key headful
+ * @bug 4217252
+ * @summary Verify that scrolling beyond the visible region and scrolling
+ *          a component smaller than the viewport is not allowed.
+ * @library /javax/swing/regtesthelpers
+ * @build Util
+ * @run main/othervm -Dsun.java2d.uiScale=1 ScrollRectToVisibleTest3
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.table.AbstractTableModel;
+
+public class ScrollRectToVisibleTest3 {
+    private static JFrame frame;
+    private static JTable table;
+    private static JButton scrollButton;
+    private static volatile int clickCount = 0;
+    private static final String[] EXPECTED_TEXT = {"99 x 0", "98 x 0",
+                                                   "97 x 0", "96 x 0"};
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        robot.setAutoWaitForIdle(true);
+
+        SwingUtilities.invokeAndWait(ScrollRectToVisibleTest3::createTestUI);
+        robot.waitForIdle();
+        robot.delay(1000);
+
+        Rectangle frameBounds = Util.invokeOnEDT(() -> getComponentBounds(frame));
+        robot.delay(100);
+        Point scrollBtnLoc = Util.getCenterPoint(scrollButton);
```

Comment:
scrollButton initialized on EDT

Reply:
[Util.getCenterPoint()](https://github.com/openjdk/jdk/blob/db2dffb6e5fed3773080581350f7f5c0bcff8f35/test/jdk/javax/swing/regtesthelpers/Util.java#L197) is invoked on EDT.

Reply:
However, scrollButton is a parameter of getCenterPoint() and is passed to the method on the main thread.